sympy/solvers/solvers.py

```diff
@@ -228,7 +228,7 @@ def checksol(f, symbol, sol=None, **flags):
                     return False
                 # there are free symbols -- simple expansion might work
                 _, val = val.as_content_primitive()
-                val = expand_mul(expand_multinomial(val))
+                val = simplify(expand_mul(expand_multinomial(val)))
```

Comment:
In my [case](https://github.com/sympy/sympy/pull/11339/files#diff-455c9876fa95231544cafe50ee8b9b09R1075) `trigsimp` is required. But it is not there.